Welcome widget timer is started even if `show_welcome_screen=False`

### 🐛 Bug Report

When running the pretest in PartSeg I spot that the overlay timer is started even if `show_welcome_screen=False` 

https://github.com/napari/napari/actions/runs/21475058637/job/61856922435?pr=8609

### 💡 Steps to Reproduce

```python
  _____________________________________ TestResultImageView.test_setting_properties ______________________________________
  
  self = <test_PartSeg.roi_analysis.test_image_view.TestResultImageView object at 0x11e1a0890>
  part_settings = <PartSeg._roi_analysis.partseg_settings.PartSettings object at 0x1291028f0>
  image2 = Image(shape=(1, 20, 20, 20) dtype=uint16, spacing=(0.001, 0.001, 0.001), labels=['channel 1', 'channel 2', 'channel 3'], channels=3, axes='CTZYX', mask=False)
  qtbot = <pytestqt.qtbot.QtBot object at 0x11ede8c50>
  
      def test_setting_properties(self, part_settings, image2, qtbot):
          prop = ChannelProperty(part_settings, "test")
  >       view = ResultImageView(part_settings, prop, "test")
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  package/tests/test_PartSeg/roi_analysis/test_image_view.py:78: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/PartSeg/_roi_analysis/image_view.py:21: in __init__
      super().__init__(settings, channel_property, name)
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/PartSeg/common_gui/napari_image_view.py:160: in __init__
      self.viewer_widget = NapariQtViewer(self.viewer)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/PartSeg/common_gui/napari_image_view.py:938: in __init__
      super().__init__(viewer, show_welcome_screen=False)
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_qt/qt_viewer.py:205: in __init__
      self.canvas = canvas_class(
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/canvas.py:217: in __init__
      self._update_viewer_overlays()
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/canvas.py:942: in _update_viewer_overlays
      vispy_overlay = create_vispy_overlay(
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/utils/visual.py:127: in create_vispy_overlay
      return overlay_to_visual[cls](overlay=overlay, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/overlays/welcome.py:43: in __init__
      self.reset()
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/overlays/welcome.py:87: in reset
      super().reset()
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/overlays/base.py:88: in reset
      super().reset()
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/overlays/base.py:48: in reset
      self._on_visible_change()
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/napari/_vispy/overlays/welcome.py:65: in _on_visible_change
      self.tip_timer.start()
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/vispy/app/timer.py:118: in start
      self._backend._vispy_start(self.interval)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = <vispy.app.backends._qt.TimerBackend object at 0x11ef09770>, interval = 10.0
  
      def _vispy_start(self, interval):
  >       self.start(int(interval * 1000))
  E       TypeError: _block_threads.<locals>.not_start() takes 1 positional argument but 2 were given
  
  /private/tmp/tox/py312-PyQt6-all/lib/python3.12/site-packages/vispy/app/backends/_qt.py:1034: TypeError
```

### 💡 Expected Behavior

if `show_welcome_screen=False` is passed to the `QtViewer` constructor, then the timer for the Welcome widget should not be started. 

### 🌎 Environment

napari 0.7.0a2

### 💡 Additional Context

_No response_

--- a/src/napari/components/overlays/welcome.py
+++ b/src/napari/components/overlays/welcome.py
@@ -7,7 +7,6 @@ from napari.components.overlays.base import CanvasOverlay
 class WelcomeOverlay(CanvasOverlay):
     """Welcome screen overlay."""
 
-    visible: bool = True
     # not settable in this specific overlay
     position: None = None
     gridded: Literal[False] = False

